src/ruby/puppet-server-lib/puppet/server/master.rb

```diff
@@ -126,14 +89,17 @@ def terminate
   private
 
   def self.getManifests(env)
-    manifests = []
-    if env.manifest != Puppet::Node::Environment::NO_MANIFEST
-      if File.directory?(env.manifest)
-        manifests = Dir.glob(File.join(env.manifest, '**/*.pp')).sort
-      else
-        manifests = [env.manifest]
+    manifests =
+      case
+        when env.manifest == Puppet::Node::Environment::NO_MANIFEST
+          []
+        when File.directory?(env.manifest)
+          Dir.glob(File.join(env.manifest, '**/*.pp'))
+        when File.exists?(env.manifest)
```

Comment:
Not sure.  Mostly just wanted to make sure that Henrik knows what the current behavior is, and that we all agree on whether or not it's the best behavior.